producer/writer/consumer_service_writer.go

```diff
@@ -172,6 +180,8 @@ func (w *consumerServiceWriterImpl) Write(rm producer.RefCountedMessage) {
 }
 
 func (w *consumerServiceWriterImpl) Init(t initType) error {
+	go w.reportMetrics()
```

Comment:
Do you want to wait for this goroutine to finish before you exit from `Close`?

Reply:
yeah sg